everything-cat/src/components/catInfo/CatBreedInfo.jsx

```diff
@@ -0,0 +1,88 @@
+import React, {useEffect, useState} from 'react';
+import {useMutation, useQuery} from '@tanstack/react-query';
+import {isCatFavourite, markCatAsFavourite} from '../../api';
+import {CardMedia, Checkbox, FormControlLabel, Typography, Box} from "@mui/material";
+import BreedInfo from "./BreedInfo.jsx";
+
+/**
+ * Component for showing the selected cat's info and image.
+ *
+ * @param catData
+ * @returns {Element}
+ * @constructor
+ */
+const CatBreedInfo = ({ catData }) => {
+    const [isFavourite, setIsFavourite] = useState(false);
+
+    const {
+        data: catIsFavourite,
+        isFetched: catIsFavouriteIsFetched,
+    } = useQuery({
+        queryKey: [`cat_${catData.id}_favourite`],
+        queryFn: () => isCatFavourite(catData.id),
+    });
+
+    useEffect(() => {
+        if (catIsFavourite) {
+            setIsFavourite(Boolean(catIsFavourite.length));
+        }
+    }, [catIsFavourite]);
```

Comment:
Do you really need this useEffect here?

Reply:
And why do you need this extra isFavourite in your component's state when you already have this information (catIsFavourite)?

Reply:
The API response `catIsFavourite` indicates if the user has marked the cat image as a favorite, while `isFavorite` is an internal component state used to toggle the checkbox. I could have named it better, like `userHasCheckedFavorite`. I designed it this way to avoid calling the `isCatFavourite` API each time the checkbox is toggled, optimizing performance.

Reply:
> Do you really need this useEffect here?

You are correct we can avoid the useEffect here if checkbox is a different component. That way we control the renders of the new checkbox component, thus it's internal state, by passing `catIsFavourite?.length` as a prop.